modules/pytket-projectq/pytket/extensions/projectq/backends/projectq_backend.py

```diff
@@ -81,8 +81,35 @@ def _result_id_type(self) -> _ResultIdTuple:
         return (str,)
 
     @property
-    def device(self) -> "Optional[Device]":
-        return None
+    def characterisation(self) -> Dict[str, Any]:
+        return dict()
+
+    @property
+    def backend_info(self) -> BackendInfo:
+        _gate_set = {
+            OpType.SWAP,
+            OpType.CRz,
+            OpType.CX,
+            OpType.CZ,
+            OpType.H,
+            OpType.X,
+            OpType.Y,
+            OpType.Z,
+            OpType.S,
+            OpType.T,
+            OpType.V,
+            OpType.Rx,
+            OpType.Ry,
+            OpType.Rz,
+            OpType.Barrier,
+        }
+        backend_info = BackendInfo(
+            "project_q_backend",
```

Comment:
```suggestion
            "projectq_backend",
```
Would it be better if all the names used here just matched the class names?

Reply:
Yes, that's a good point, although I am not sure in this case which one would be correct, given the spelling `ProjectQBackend`. Should I go with `projectq_backend` to match the filename and extension name, or `project_q_backend` to have a direct correspondence to CamelCase?

Reply:
My instinct would be to exactly match the class name in all cases. It might even be possible to do that automatically by introspection?

Reply:
Or maybe fully qualified, as in `pytket.extensions.qiskit.backends.aer.AerBackend`. This seems to match what mushroom are doing now, However, I'm a little resistant to that as it depends on knowledge of how the python is packaged.

Reply:
Ok, thanks. using the class names sounds sensible for now. We can easily change this later if necessary.